File: app/src/main/java/com/otraempresa/android/example/RequestTokenChargeAsyncTask.java
package com.otraempresa.android.example;

import android.content.Context;

import com.kushkipagos.android.Card;
import com.kushkipagos.android.KushkiException;
import com.kushkipagos.android.Transaction;

class RequestTokenChargeAsyncTask extends AbstractRequestTokenChargeAsyncTask {
    Context appContext;
    Card card = new Card("Patricio Moreaon","4242424242424242","111","06","22");

    RequestTokenChargeAsyncTask(Context context) {
        super(context);
        this.appContext=context;
    }

    @Override
    protected Transaction requestTokenCharge(String params) throws KushkiException {
        return kushki.requestTokenCharge(card,10,params,true,appContext);
    }

}

File: app/src/main/java/com/kushkipagos/kushkidemo/services/RequestTokenChargeAsyncTask.kt
package com.kushkipagos.kushkidemo.services

import android.content.Context
import com.kushkipagos.exceptions.KushkiException
import com.kushkipagos.kushkidemo.contracts.AbstractRequestTokenChargeAsyncTask
import com.kushkipagos.models.Card
import com.kushkipagos.models.Transaction

class RequestTokenChargeAsyncTask: AbstractRequestTokenChargeAsyncTask {
    val _context: Context

    var card = Card("Patricio Moreaon", "4242424242424242", "111", "06", "22")

    constructor(context: Context): super(context){
        _context = context
    }

    @Throws(KushkiException::class)
    override fun requestTokenCharge(params: String): Transaction {
        return _kushki.requestTokenCharge(card, 10.0, params, true, _context)
    }
}